### PR TITLE
Update 833.isb

### PR DESCRIPTION
**Pull request by porcupyn**
Added missing stanza and corrected a few typos. Have a couple of questions.

BTW, I made a couple of other changes today, but don't think I've checked them back in properly. Vinay, can you review please?

### CODE REVIEW

**Comment by morosebose** [comment]
Hey @porcupyn, good to see your contributions! Could you please follow these suggestions for best practices?

1. Create a new pull request for each song. Don't bundle multiple songs into the same pull request. 
2. Instead of working off master, create a new branch for each song that you want to update. That way, we won't have multiple folks trying to update the master branch at once, in case you're working on a song and I'm working on a different song.

Hope this is helpful!

File: docs/833.isb.txt [hunk 1]
```diff
@@ -15,19 +15,30 @@
 \printtitle 
 #indian
 %
-jab tak pUre naa ho.n phere saat 
+babuwaa ho babuwaa, pahunaa ho pahunaa
+babuwaa ho babuwaa, pahunaa ho pahunaa
+
+jab tak pUre naa ho.n phere saat
```
Comment (morosebose) [inline docs/833.isb.txt]: puure

File: docs/833.isb.txt [hunk 2]
```diff
@@ -15,19 +15,30 @@
...
 tab tak dulhan nahii.n dulhaa kii
```
Comment (morosebose) [inline docs/833.isb.txt]: dulhin

File: docs/833.isb.txt [hunk 3]
```diff
@@ -15,19 +15,30 @@
...
+
+jab tak pUre naa ho.n phere saat
 tab tak dulhan nahii.n dulhaa kii 
 re tab tak babunii nahii.n babuvaa kii, naa, jab tak \threedots 
 
-1) abhahii to pahunaa pahalii bha.nvar pa.Dii hai
-abhii.n to dillii dUr kha.Dii hai
-ho pahalii bha.nvar pa.Dii hai dillii dUr kha.Dii hai
-saat phere saat janmo.n kaa saath, jab tak pUre naa \threedots 
+1) ab hii to babuwaa pahalii hii bha.nvar pa.Dii hai
```
Comment (morosebose) [inline docs/833.isb.txt]: Remove the stanza number. Also, "bha.Nvar"

File: docs/833.isb.txt [hunk 4]
```diff
@@ -15,19 +15,30 @@
...
+ab hii to pahunaa dillii hii dUr ba.Dii hai
```
Comment (morosebose) [inline docs/833.isb.txt]: duur

File: docs/833.isb.txt [hunk 5]
```diff
@@ -15,19 +15,30 @@
...
+ho pahalii hii bha.nvar pa.Dii hai, dillii dUr ba.Dii hai
```
Comment (morosebose) [inline docs/833.isb.txt]: bha.Nvar ... duur

File: docs/833.isb.txt [hunk 6]
```diff
@@ -15,19 +15,30 @@
...
+karanii hogii tapasyaa saarii raat, jab tak pUre naa \threedots
```
Comment (morosebose) [inline docs/833.isb.txt]: puure

File: docs/833.isb.txt [hunk 7]
```diff
@@ -15,19 +15,30 @@
...
+ho pahalii hii bha.nvar pa.Dii hai, dillii dUr ba.Dii hai
+karanii hogii tapasyaa saarii raat, jab tak pUre naa \threedots 
 
-2) jaise jaise bha.Nvar pa.De man apano.n ko chho.De 
+2) jaise jaise bha.Nvar pa.De man a.Nganaa ko chho.De 
 ek ek bhaa.Nvar naataa anjaano.n se jo.De 
-man ghar apano.n ko chho.De, anjaano.n se naataa jo.De 
+man ghar a.Nganaa ko chho.De, anjaano.n se naataa jo.De 
 sukh kii badarii aa.Nsuu kii barasaat, jab tak pUre naa \threedots 
+
```
Comment (morosebose) [inline docs/833.isb.txt]: puure

File: docs/833.isb.txt [hunk 8]
```diff
@@ -15,19 +15,30 @@
...
+babuwaa ho babuwaa, pahunaa ho pahunaa
+saat phere karo babuwaa bharo saat vachan bhii
+aise(?) kanyaa aise(?) arpan kar de tan bhii man bhii
```
Comment (morosebose) [inline docs/833.isb.txt]: aise kanyaa kaise arpaN kar de.n tan bhii man bhii

File: docs/833.isb.txt [hunk 9]
```diff
@@ -15,19 +15,30 @@
...
+uTho uTho babunii dekho dekho dhruv taaraa
+dhruv taare saa ho amar suhaag tihaaraa
+ho dekho dekho dhruv taaraa, amar suhaag tihaaraa
+saato.n phere saat janmo.n kaa saath, jab tak pUre naa \threedots
```
Comment (morosebose) [inline docs/833.isb.txt]: puure

**Comment by morosebose** [comment]
Whoops, also remove stanza number for the second stanza. 

Sorry, if this were a separate branch I'd have downloaded this and just done a fresh commit/PR but I don't know how to handle an unmerged commit into master.

**Comment by morosebose** [comment]
Oh wait I just realized these changes were on porcupyn:master and not the actual v9y:master, duh. Sorry about that.

File: docs/833.isb.txt (hunk 9 above)
Comment (porcupyn) [reply docs/833.isb.txt]: Sorry guys! I am still learning this github thing. And I still don't know if I am doing this right or not.

So, I have a "porcupyn" version which is a copy of the v9y version, right? Do I start off of that version or of the main version? I am about to add a paragraph to the song "apani aankhon men basaakar"...

Also in response to Surajit, do I need to modify existing lyrics that appear correct and are transliterating properly? For example, dUr vs. duur - the song had dUr from before and it was working fine...

**Comment by porcupyn** [comment]
Also, I disagree with the correction of dulhan to dulian (sic). The correct word is dulhan whether or not the singer sings it as dulhin, no? Else, we would have to change 'pahali' to 'peheli' etc.

**Comment by v9y** [comment]
dulian was a typo (I don't know how that happened). Fixed that. Thanks for pointing out.

In both speech and written forms 'dulhin' is a valid variant and is often used. No one writes pahali as peheli (or even speaks it that way - the slight laghu 'e'-ish sound that many people use before 'ha' and not 'li' - This is more of a general pronunciation rule in Hindi - the sound of 'a' changes slightly to laghu 'e' if it is followed by 'ha').  In cases, where there are multiple accepted variants, we should try to approximate the spoken version in transcribing.

**Comment by porcupyn** [comment]
Cool! I've been so long outside India that I've really got to think hard to
figure out if I've even heard "dulhin" or "dulhan" more often! And I'm
unsuccessful in answering that question!

BTW, did you see my other pull request? Have I done that properly?

Thanks,
- Sudarshan

**Comment by v9y** [comment]
Sudarshan: I don't see any other PR pending. Can you check and try submitting again? Or please send me the URL if you see it.

**Comment by porcupyn** [comment]
I did something (merge) just now - do you see it now? It is for isb209

**Comment by morosebose** [comment]
Porky,

You likely don't have merge access to the main repo. Are you issuing pull requests against Vinay's master or your local copy? We won't see requests made only against your copy.

- Surajit

**Comment by porcupyn** [comment]
I bet that's what I'm doing. :-(

Like I've said, I'm learning github. How exactly does it work? In the
version control we have at work, I don't have a local version (like I
appear to be doing for the giitaayan files @ github). But I thought I was
supposed to do that for giitaayan, or did I misunderstand?

Also, a question about FITB - is there a specific sequence I can use? If
not, how do I know the latest number already used?

**Comment by morosebose** [comment]
There isn't a good reason to maintain your own clone of the giitayan repo. As long as you have permissions to create a branch off of the master branch of Vinay's repo, you can make changes to that branch via commits. When the changes look good, you can issue a pull request. At that time, Vinay (or anybody else with permissions) can look at the changes you've made, make comments and/or other commits, and (once everyone is satisfied) pull those changes into the master branch. After that, you can delete the branch you had created as all its changes are now in master. 

Make sense?
- Surajit

**Comment by porcupyn** [comment]
Makes sense - thanks - but it appears that I created a clone when I was
creating a branch. Or did I create a branch which I now think is a clone?
How do I tell them apart?

**Comment by morosebose** [comment]
Branches are off a repo.

A repo can be cloned, which duplicates the repo in a different user space. That cloned repo, in turn, can have its own branches. Those will not propagate to the original repo. 

To make changes to *any given* repo, create a branch, make commits against that branch, then issue a pull request to the owner of the repo.

If you are creating branches and making commits against your clone, rather than against Vinay's original, then changes you make won't make it to giitaayan, just to your clone. You need to create branches, make commits, and issue pull requests against Vinay's original repo.

(The above is a simplification, but it seems like the most helpful way to clarify things atm.)

- s

**Comment by v9y** [comment]
Surajit explained the whole concept well. Hope that helps. Another way to look at it: 

Main Repo (v9y/giit) 
 - Forked Repo (porcupyn/giit) 
    - [Refresh your repo periodically from v9y/giit so that you get the changes made by others]
    - Create a branch or feel free to work directly on the porcupyn:master branch (since this is your repo and you are the only one using it) 
       - [Optional] Clone forked repo locally on your machine. You can also instead work directly on github
          - Repeat as many times as needed:
            - Make changes 
            - Commit changes 
       - If you worked locally then push your changes back to your repo. Ignore if worked on github.
   - Create a pull request to merge your master into main repo's master

**Comment by porcupyn** [comment]
Thanks Vinay and Surajit!

I understand (I think) now, but I am confused about one aspect.

It appears that the best and easiest way to make changes is to get a branch
off the main repo and make changes then create a pull request. For some
reason, I had the notion that I needed to create a forked repo. That is my
misunderstanding, right? Or is that part of the best practices guidelines
for giitaayan?

**Comment by morosebose** [comment]
Generally the reason to fork a repo is if you want to make changes to a codebase that are not necessarily going to make it back into the original. E.g., if you use an open source base but make proprietary changes to it, or if you're just experimenting with the code for your own purposes, or to learn. Or if you just want a redundant copy for safety/security reasons. 

I did fork Vinay's original repo but eventually just cloned it to my local machine as I prefer to work in a text editor rather than a web interface. I update the fork from time to time to keep it current. 

You asked about FITB serial numbers. Right now I just know that we've had 152 entries (including jab tak pure na ho) because I keep a list locally. I will publish it soon, I'm behind on a couple of ISB related projects (including some coding for Vinay around sequence numbers). 

- s

**Pull request merged**